diff --git a/CMakeLists.txt b/CMakeLists.txt
index 374dec84..2c435603 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -601,7 +601,6 @@ if(DEMANGLE_DIR)
 endif()
 
 ## Set RCCL compile options
-target_compile_options(rccl PRIVATE -parallel-jobs=12)
 target_compile_options(rccl PRIVATE -Wno-format-nonliteral)
 target_compile_options(rccl PRIVATE -fgpu-rdc)               # Generate relocatable device code (required for extern __shared__)
 target_compile_options(rccl PRIVATE -fvisibility=hidden)     # Set symbol visibility to hidden by default
@@ -634,7 +633,6 @@ endif()
 target_link_libraries(rccl PRIVATE -fgpu-rdc)               # Required when linking relocatable device code
 
 ## Set RCCL link options
-target_link_options(rccl PRIVATE -parallel-jobs=16)       # Use multiple threads to link
 if(BUILD_ADDRESS_SANITIZER)
   target_link_options(rccl PRIVATE -fuse-ld=lld)
 endif()
diff --git a/src/include/gdrwrap.h b/src/include/gdrwrap.h
index f532a705..635bce22 100644
--- a/src/include/gdrwrap.h
+++ b/src/include/gdrwrap.h
@@ -41,7 +41,7 @@ static inline void wc_store_fence(void) { asm volatile("sync") ; }
 #elif defined(__x86_64__)
 #include <immintrin.h>
 static inline void wc_store_fence(void) { _mm_sfence(); }
-#elif defined(__aarch64__)
+#elif defined(__aarch64__) || defined(__loongarch64)
 #ifdef __cplusplus
 #include <atomic>
 static inline void wc_store_fence(void) { std::atomic_thread_fence(std::memory_order_release); }
