diff --git a/clients/benchmarks/CMakeLists.txt b/clients/benchmarks/CMakeLists.txt
index a8aea885b..273ed5244 100644
--- a/clients/benchmarks/CMakeLists.txt
+++ b/clients/benchmarks/CMakeLists.txt
@@ -143,15 +143,6 @@ else( )
   endif()
 endif()
 
-if( CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  # GCC or hip-clang needs specific flags to turn on f16c intrinsics
-  target_compile_options( rocblas-bench PRIVATE -mf16c )
-  target_compile_options( rocblas_v3-bench PRIVATE -mf16c )
-  if( BUILD_WITH_TENSILE )
-    target_compile_options( rocblas-gemm-tune PRIVATE -mf16c )
-  endif()
-endif()
-
 target_compile_definitions( rocblas-bench PRIVATE ROCBLAS_BENCH ROCM_USE_FLOAT16 ROCBLAS_INTERNAL_API ROCBLAS_NO_DEPRECATED_WARNINGS ${TENSILE_DEFINES} )
 target_compile_definitions( rocblas_v3-bench PRIVATE ROCBLAS_BENCH ROCM_USE_FLOAT16 ROCBLAS_INTERNAL_API ROCBLAS_NO_DEPRECATED_WARNINGS ${TENSILE_DEFINES} )
 if( BUILD_WITH_TENSILE )
diff --git a/clients/gtest/CMakeLists.txt b/clients/gtest/CMakeLists.txt
index 55e9bbe5b..cacef5174 100644
--- a/clients/gtest/CMakeLists.txt
+++ b/clients/gtest/CMakeLists.txt
@@ -202,12 +202,6 @@ else( )
   target_link_libraries( rocblas_v3-test PRIVATE hip::host hip::device )
 endif( )
 
-if( CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  # GCC or hip-clang needs specific flag to turn on f16c intrinsics
-  target_compile_options( rocblas-test PRIVATE -mf16c )
-  target_compile_options( rocblas_v3-test PRIVATE -mf16c )
-endif( )
-
 target_compile_definitions( rocblas-test PRIVATE ROCM_USE_FLOAT16 ROCBLAS_INTERNAL_API ROCBLAS_NO_DEPRECATED_WARNINGS )
 target_compile_definitions( rocblas_v3-test PRIVATE ROCM_USE_FLOAT16 ROCBLAS_INTERNAL_API ROCBLAS_NO_DEPRECATED_WARNINGS )
 
diff --git a/clients/samples/CMakeLists.txt b/clients/samples/CMakeLists.txt
index 588020c93..dcd512a59 100644
--- a/clients/samples/CMakeLists.txt
+++ b/clients/samples/CMakeLists.txt
@@ -106,7 +106,6 @@ foreach( exe ${sample_list_all} )
 
   if( CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     # GCC or hip-clang needs specific flags to turn on f16c intrinsics
-    target_compile_options( ${exe} PRIVATE -mf16c )
     target_compile_definitions( ${exe} PRIVATE ROCBLAS_INTERNAL_API )
   endif( )
 
diff --git a/library/src/CMakeLists.txt b/library/src/CMakeLists.txt
index 5594284b3..e982d61cc 100755
--- a/library/src/CMakeLists.txt
+++ b/library/src/CMakeLists.txt
@@ -421,9 +421,6 @@ endif()
 
 #  -fno-gpu-rdc compiler option was used with hcc, so revisit feature at some point
 
-# GCC or hip-clang needs specific flags to turn on f16c intrinsics
-target_compile_options( rocblas PRIVATE -mf16c )
-
 # Do not allow Variable Length Arrays (use unique_ptr instead)
 target_compile_options( rocblas PRIVATE -Werror=vla )
 
