diff --git a/src/queues.c b/src/queues.c
index b800681..5d5254c 100644
--- a/src/queues.c
+++ b/src/queues.c
@@ -328,10 +328,14 @@ void *allocate_exec_aligned_memory_gpu(uint32_t size, uint32_t align,
 	flags.ui32.PageSize = HSA_PAGE_SIZE_4KB;
 	flags.ui32.CoarseGrain = DeviceLocal;
 	flags.ui32.Uncached = Uncached;
+#if defined(__loongarch64)
+    flags.ui32.NoNUMABind = 1;
+#endif
 
 	/* Get the closest cpu_id to GPU NodeId for system memory allocation
 	 * nonPaged=1 system memory allocation uses GTT path
 	 */
+#if !defined(__loongarch64)
 	if (!DeviceLocal && !nonPaged) {
 		cpu_id = get_direct_link_cpu(NodeId);
 		if (cpu_id == INVALID_NODEID) {
@@ -340,6 +344,7 @@ void *allocate_exec_aligned_memory_gpu(uint32_t size, uint32_t align,
 		}
 	}
 
+#endif
 	size = ALIGN_UP(size, align);
 
 	ret = hsaKmtAllocMemory(DeviceLocal ? NodeId : cpu_id, size, flags, &mem);
