diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 8714e11..84a51d3 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -84,7 +84,7 @@ find_file(HIP_RUNTIME_API_H hip_runtime_api.h
 add_custom_command(
   OUTPUT hip_ostream_ops.h
   COMMAND ${CMAKE_C_COMPILER} "$<$<BOOL:${HIP_INCLUDE_DIRECTORIES}>:-I$<JOIN:${HIP_INCLUDE_DIRECTORIES},$<SEMICOLON>-I>>"
-          -E "${HIP_RUNTIME_API_H}" -D__HIP_PLATFORM_HCC__=1 -D__HIP_ROCclr__=1 -o hip_runtime_api.h.i
+          -E "${HIP_RUNTIME_API_H}" -D__HIP_PLATFORM_HCC__=1 -D__HIP_ROCclr__=1 -D__HIP_PLATFORM_AMD__=1 -o hip_runtime_api.h.i
   BYPRODUCTS hip_runtime_api.h.i
   COMMAND ${Python3_EXECUTABLE} ${PROJECT_SOURCE_DIR}/script/gen_ostream_ops.py
           -in hip_runtime_api.h.i -out hip_ostream_ops.h > /dev/null
