diff --git a/CMakeLists.txt b/CMakeLists.txt
index c36d936e..f97a80b4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -196,8 +196,9 @@ else()
       virtualenv_install(${Tensile_TEST_LOCAL_PATH})
       message (STATUS "using local Tensile from ${Tensile_TEST_LOCAL_PATH}, copied to ${Tensile_ROOT}")
     else()
-      virtualenv_install(${CMAKE_SOURCE_DIR}/tensilelite)
+      virtualenv_install(${CMAKE_SOURCE_DIR}/tensilelite joblib /home/xinmu/rocm-loongarch/depend/orjson/orjson/target/wheels/orjson-3.9.15-cp310-cp310-linux_loongarch64.whl msgpack)
     endif()
+
     message(STATUS "Adding ${VIRTUALENV_HOME_DIR} to CMAKE_PREFIX_PATH")
     list(APPEND CMAKE_PREFIX_PATH ${VIRTUALENV_HOME_DIR})
     if (TENSILE_VERSION)
diff --git a/tensilelite/Tensile/Ops/gen_assembly.sh b/tensilelite/Tensile/Ops/gen_assembly.sh
index 0b21b6c6..ed992235 100755
--- a/tensilelite/Tensile/Ops/gen_assembly.sh
+++ b/tensilelite/Tensile/Ops/gen_assembly.sh
@@ -28,12 +28,12 @@ dst=$2
 venv=$3
 build_id_kind=$4
 
-rocm_path=/opt/rocm
+rocm_path=/opt/rocm-next
 if ! [ -z ${ROCM_PATH+x} ]; then
     rocm_path=${ROCM_PATH}
 fi
 
-toolchain=${rocm_path}/llvm/bin/clang++
+toolchain=${rocm_path}/bin/clang++
 
 . ${venv}/bin/activate
 
diff --git a/tensilelite/Tensile/Source/lib/CMakeLists.txt b/tensilelite/Tensile/Source/lib/CMakeLists.txt
index 6111787f..68e64728 100644
--- a/tensilelite/Tensile/Source/lib/CMakeLists.txt
+++ b/tensilelite/Tensile/Source/lib/CMakeLists.txt
@@ -102,11 +102,11 @@ if(TENSILE_USE_LLVM OR TENSILE_USE_MSGPACK)
 endif()
 
 if(TENSILE_USE_MSGPACK)
-    find_package(msgpack REQUIRED)
+    find_package(msgpack-cxx REQUIRED)
     target_compile_definitions(TensileHost PUBLIC -DTENSILE_MSGPACK=1)
 
-    if(TARGET msgpackc-cxx)
-        get_target_property(msgpack_inc msgpackc-cxx INTERFACE_INCLUDE_DIRECTORIES)
+    if(TARGET msgpack-cxx)
+        get_target_property(msgpack_inc msgpack-cxx INTERFACE_INCLUDE_DIRECTORIES)
     elseif(TARGET msgpackc)
         get_target_property(msgpack_inc msgpackc INTERFACE_INCLUDE_DIRECTORIES)
     endif()
diff --git a/tensilelite/requirements.txt b/tensilelite/requirements.txt
index f8ac0f62..ce73edf6 100644
--- a/tensilelite/requirements.txt
+++ b/tensilelite/requirements.txt
@@ -4,4 +4,3 @@ msgpack
 joblib
 simplejson
 ujson
-orjson
