diff --git a/Configure.cmake b/Configure.cmake
index facd1f8..5444b71 100644
--- a/Configure.cmake
+++ b/Configure.cmake
@@ -235,6 +235,14 @@ elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "s390x")
 
   set(TESTER3_DEFINITIONS_ZVECTOR2      ATR=finz_ DPTYPE=SLEEF_VECTOR_DOUBLE SPTYPE=SLEEF_VECTOR_FLOAT DPTYPESPEC=d2 SPTYPESPEC=f4 EXTSPEC=zvector2)
   set(TESTER3_DEFINITIONS_ZVECTOR2NOFMA ATR=cinz_ DPTYPE=SLEEF_VECTOR_DOUBLE SPTYPE=SLEEF_VECTOR_FLOAT DPTYPESPEC=d2 SPTYPESPEC=f4 EXTSPEC=zvector2nofma)
+else()
+
+  set(SLEEF_HEADER_LIST
+    PUREC_SCALAR
+    PURECFMA_SCALAR)
+
+  set(CLANG_FLAGS_ENABLE_PURECFMA_SCALAR )
+
 endif()
 
 command_arguments(HEADER_PARAMS_PUREC_SCALAR    cinz_ 1 1 double float int32_t int32_t __STDC__ purec)
diff --git a/src/arch/helperpurec_scalar.h b/src/arch/helperpurec_scalar.h
index 10e8602..5b1085b 100644
--- a/src/arch/helperpurec_scalar.h
+++ b/src/arch/helperpurec_scalar.h
@@ -54,7 +54,7 @@
 #define ENABLE_FMA_SP
 //@#define ENABLE_FMA_SP
 
-#if defined(__AVX2__) || defined(__aarch64__) || defined(__arm__) || defined(__powerpc64__) || defined(__zarch__)
+#if defined(__AVX2__) || defined(__aarch64__) || defined(__arm__) || defined(__powerpc64__) || defined(__zarch__) || defined(__loongarch_lp64)
 #ifndef FP_FAST_FMA
 #define FP_FAST_FMA
 //@#define FP_FAST_FMA
diff --git a/src/libm-tester/iutsimd.c b/src/libm-tester/iutsimd.c
index aa97f66..a647b39 100644
--- a/src/libm-tester/iutsimd.c
+++ b/src/libm-tester/iutsimd.c
@@ -31,7 +31,7 @@
 #include <float.h>
 #include <limits.h>
 
-#if defined(__AVX2__) || defined(__aarch64__) || defined(__arm__) || defined(__powerpc64__)
+#if defined(__AVX2__) || defined(__aarch64__) || defined(__arm__) || defined(__powerpc64__) || defined(__loongarch_lp64)
 #ifndef FP_FAST_FMA
 #define FP_FAST_FMA
 #endif
