diff --git a/src/HipifyAction.cpp b/src/HipifyAction.cpp
index 2c635d58..b7b5c42a 100644
--- a/src/HipifyAction.cpp
+++ b/src/HipifyAction.cpp
@@ -1164,8 +1164,8 @@ void HipifyAction::EndSourceFileAction() {
     clang::Preprocessor &PP = getCompilerInstance().getPreprocessor();
     clang::HeaderSearch &HS = PP.getHeaderSearchInfo();
     clang::ExternalPreprocessorSource *EPL = HS.getExternalLookup();
-    const clang::FileEntry *FE = SM.getFileEntryForID(SM.getMainFileID());
-    const clang::IdentifierInfo *controllingMacro = HS.getFileInfo(FE).getControllingMacro(EPL);
+    const clang::FileEntryRef FER = SM.getFileEntryRefForID(SM.getMainFileID()).value();
+    const clang::IdentifierInfo *controllingMacro = HS.getFileInfo(FER).getControllingMacro(EPL);
     if (controllingMacro) {
       auto found = Ifndefs.find(controllingMacro->getName().str());
       if (found != Ifndefs.end()) {
