diff --git a/include/ck/ck.hpp b/include/ck/ck.hpp
index 32eea551f..325d34c62 100644
--- a/include/ck/ck.hpp
+++ b/include/ck/ck.hpp
@@ -21,7 +21,11 @@ CK_DECLARE_ENV_VAR_BOOL(CK_LOGGING)
 
 // constant address space for kernel parameter
 // https://llvm.org/docs/AMDGPUUsage.html#address-spaces
+#if !defined(__loongarch_lp64)
 #define CK_CONSTANT_ADDRESS_SPACE __attribute__((address_space(4)))
+#else
+#define CK_CONSTANT_ADDRESS_SPACE
+#endif
 
 // launch bounds
 #define CK_USE_LAUNCH_BOUNDS 1
