diff --git a/rocclr/device/rocm/rocvirtual.cpp b/rocclr/device/rocm/rocvirtual.cpp
index ed4649998..541f6f40f 100644
--- a/rocclr/device/rocm/rocvirtual.cpp
+++ b/rocclr/device/rocm/rocvirtual.cpp
@@ -43,7 +43,9 @@
 #include <string>
 #include <thread>
 #include <vector>
+#ifdef __x86_64
 #include <immintrin.h>
+#endif
 
 
 /**
@@ -2789,7 +2791,9 @@ bool VirtualGPU::createVirtualQueue(uint deviceQueueSize)
 
   return true;
 }
-
+#if !defined(__x86_64)
+#include <string.h>
+#endif
 // ================================================================================================
 __attribute__((optimize("unroll-all-loops"), always_inline))
 static inline void nontemporalMemcpy(void* __restrict dst, const void* __restrict src,
@@ -2809,6 +2813,7 @@ static inline void nontemporalMemcpy(void* __restrict dst, const void* __restric
     }
     size = size % sizeof(__m256i);
   #endif
+#if defined(__x86_64)
 
   for (auto i = 0u; i != size / sizeof(__m128i); ++i) {
     _mm_stream_si128(reinterpret_cast<__m128i* __restrict&>(dst)++,
@@ -2826,6 +2831,10 @@ static inline void nontemporalMemcpy(void* __restrict dst, const void* __restric
     _mm_stream_si32(reinterpret_cast<int* __restrict&>(dst)++,
                     *reinterpret_cast<const int* __restrict&>(src)++);
   }
+#else
+   memcpy(dst, src, size);
+#endif
+
 }
 
 // ================================================================================================
diff --git a/rocclr/os/os_posix.cpp b/rocclr/os/os_posix.cpp
index e48a4bcfc..d733b15bf 100644
--- a/rocclr/os/os_posix.cpp
+++ b/rocclr/os/os_posix.cpp
@@ -307,8 +307,10 @@ void Os::currentStackInfo(address* base, size_t* size) {
 
   ::pthread_attr_destroy(&threadAttr);
 
+#if !defined(__loongarch64)
   assert(Os::currentStackPtr() >= *base - *size && Os::currentStackPtr() < *base &&
          "just checking");
+#endif
 }
 
 void Os::setCurrentThreadName(const char* name) { ::prctl(PR_SET_NAME, name); }
