diff --git a/rbuild.ini b/rbuild.ini
index 3eb2fef62..259294b62 100755
--- a/rbuild.ini
+++ b/rbuild.ini
@@ -1,6 +1,6 @@
 [main]
-cxx = ${rocm_path}/llvm/bin/clang++
-cc = ${rocm_path}/llvm/bin/clang
+cxx = ${rocm_path}/bin/clang++
+cc = ${rocm_path}/bin/clang
 deps =
     ROCmSoftwarePlatform/rocm-recipes
     -f requirements.txt
@@ -19,8 +19,8 @@ define =
     BUILD_DEV=On
 
 [develop]
-cxx = ${rocm_path}/llvm/bin/clang++
-cc = ${rocm_path}/llvm/bin/clang
+cxx = ${rocm_path}/bin/clang++
+cc = ${rocm_path}/bin/clang
 deps =
     -f dev-requirements.txt
     oneapi-src/oneDNN@v1.7
@@ -31,8 +31,8 @@ define =
     BUILD_DEV=On
 
 [cibuild]
-cxx = ${rocm_path}/llvm/bin/clang++
-cc = ${rocm_path}/llvm/bin/clang
+cxx = ${rocm_path}/bin/clang++
+cc = ${rocm_path}/bin/clang
 deps =
     -f dev-requirements.txt
 define =
