diff --git a/src/libhsakmt.ver b/src/libhsakmt.ver
index 15c2916..c04cefe 100644
--- a/src/libhsakmt.ver
+++ b/src/libhsakmt.ver
@@ -81,6 +81,7 @@ hsaKmtWaitOnEvent_Ext;
 hsaKmtWaitOnMultipleEvents_Ext;
 hsaKmtReplaceAsanHeaderPage;
 hsaKmtReturnAsanHeaderPage;
+hsaKmtGetAMDGPUDeviceHandle;
 
 local: *;
 };
diff --git a/src/queues.c b/src/queues.c
index f72f6d6..a7b9409 100644
--- a/src/queues.c
+++ b/src/queues.c
@@ -134,7 +134,9 @@ static void get_doorbell_map_info(uint32_t node_id,
 	uint32_t gfxv = get_gfxv_by_node_id(node_id);
 	doorbell->use_gpuvm = (is_dgpu && gfxv != GFX_VERSION_TONGA);
 	doorbell->size = DOORBELLS_PAGE_SIZE(DOORBELL_SIZE(gfxv));
-
+    if (doorbell->size < (uint32_t) PAGE_SIZE) {
+        doorbell->size = PAGE_SIZE;
+    }
 	return;
 }
 
