diff --git a/src/queues.c b/src/queues.c
index f62e862..b6546e7 100644
--- a/src/queues.c
+++ b/src/queues.c
@@ -134,7 +134,9 @@ static void get_doorbell_map_info(uint32_t node_id,
 	uint32_t gfxv = get_gfxv_by_node_id(node_id);
 	doorbell->use_gpuvm = (is_dgpu && gfxv != GFX_VERSION_TONGA);
 	doorbell->size = DOORBELLS_PAGE_SIZE(DOORBELL_SIZE(gfxv));
-
+    if (doorbell->size < (uint32_t) PAGE_SIZE) {
+        doorbell->size = PAGE_SIZE;
+    }
 	return;
 }
 
