diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9759e74c..8cdee314 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -202,7 +202,7 @@ target_sources( ${CORE_RUNTIME_TARGET} PRIVATE ${SRCS} )
 add_subdirectory( ${CMAKE_CURRENT_SOURCE_DIR}/core/runtime/trap_handler )
 add_dependencies( ${CORE_RUNTIME_TARGET} amd_trap_handler_v2 )

-if ( NOT DEFINED IMAGE_SUPPORT AND CMAKE_SYSTEM_PROCESSOR MATCHES "i?86|x86_64|amd64|AMD64" )
+if ( NOT DEFINED IMAGE_SUPPORT AND CMAKE_SYSTEM_PROCESSOR MATCHES "i?86|x86_64|amd64|AMD64|loongarch64" )
   set ( IMAGE_SUPPORT ON )
 endif()
 set ( IMAGE_SUPPORT ${IMAGE_SUPPORT} CACHE BOOL "Build with image support (default: ON for x86, OFF elsewise)." )
diff --git a/src/image/util.h b/src/image/util.h
index 8c0faaa..1f1cbad 100644
--- a/src/image/util.h
+++ b/src/image/util.h
@@ -74,6 +74,7 @@ namespace image {
 #include "mm_malloc.h"
 #if defined(__i386__) || defined(__x86_64__)
 #include <x86intrin.h>
+#elif defined(__loongarch64)
 #else
 #error                                                                                             \
     "Processor not identified.  " \
diff --git a/src/inc/hsa.h b/src/inc/hsa.h
index a70fd0f..b65cb0e 100644
--- a/src/inc/hsa.h
+++ b/src/inc/hsa.h
@@ -80,7 +80,7 @@
 // Try to detect CPU endianness
 #if !defined(LITTLEENDIAN_CPU) && !defined(BIGENDIAN_CPU)
 #if defined(__i386__) || defined(__x86_64__) || defined(_M_IX86) || \
-    defined(_M_X64)
+    defined(_M_X64) || defined(__loongarch64)
 #define LITTLEENDIAN_CPU
 #endif
 #endif
